python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -605,6 +621,12 @@ def schedule_node_termination(
             aggregate=operator.add,
         )
         self.nodes_to_terminate.append(node_id)
+        # Scale down entire replica of node being deleted
+        tags = self.provider.node_tags(node_id)
+        if TAG_RAY_REPLICA_INDEX in tags:
+            replica_id = tags[TAG_RAY_REPLICA_INDEX]
+            if replica_id not in self.replicas_to_delete:
```

Comment:
How about making `replicas_to_delete` a set instead of a list if we plan to look it up?

Reply:
Done.